From 3905bc49a072fb99916ffea17d9b970f20b7c224 Mon Sep 17 00:00:00 2001
From: Owen W. Taylor <otaylor@fishsoup.net>
Date: Fri, 30 Apr 2010 19:27:35 -0400
Subject: [PATCH] Mark MESA_swap_control and SGI_video_sync as not "direct only"

With DRI2, MESA_swap_control and SGI_video_sync are done on the
X server side, so shouldn't be marked direct_only - they only
can be supported if the server supports them.

This fix is not completely right because with DRI1, which is still
supported in some of the drivers, these are in fact direct_only
extensions and could conceivably not be advertised by the server.
---
 src/glx/glxextensions.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/glx/glxextensions.c b/src/glx/glxextensions.c
index e58c296..56239e5 100644
--- a/src/glx/glxextensions.c
+++ b/src/glx/glxextensions.c
@@ -98,7 +98,7 @@ static const struct extension_info known_glx_extensions[] = {
    { GLX(MESA_swap_control),           VER(0,0), N, N, N, N },
    { GLX(MESA_swap_frame_usage),       VER(0,0), N, N, N, N },
 #else
-   { GLX(MESA_swap_control),           VER(0,0), Y, N, N, Y },
+   { GLX(MESA_swap_control),           VER(0,0), Y, N, N, N },
    { GLX(MESA_swap_frame_usage),       VER(0,0), Y, N, N, Y },
 #endif
    { GLX(NV_float_buffer),             VER(0,0), N, N, N, N },
@@ -117,7 +117,7 @@ static const struct extension_info known_glx_extensions[] = {
    { GLX(OML_sync_control),            VER(0,0), Y, N, N, Y },
    { GLX(SGI_make_current_read),       VER(1,3), Y, N, N, N },
    { GLX(SGI_swap_control),            VER(0,0), Y, N, N, N },
-   { GLX(SGI_video_sync),              VER(0,0), Y, N, N, Y },
+   { GLX(SGI_video_sync),              VER(0,0), Y, N, N, N },
 #endif
    { GLX(SGIS_blended_overlay),        VER(0,0), N, N, N, N },
    { GLX(SGIS_color_range),            VER(0,0), N, N, N, N },
-- 
1.7.0.1

